frontend/packages/console-shared/src/components/spotlight/Spotlight.tsx

```diff
@@ -7,9 +7,22 @@ type SpotlightProps = {
   interactive?: boolean;
 };
 
+type HasHiddenAncestor = (el: Element) => boolean;
+
 const Spotlight: React.FC<SpotlightProps> = ({ selector, interactive }) => {
+  const hasHiddenAncestor: HasHiddenAncestor = (el) => {
+    if (!el) return false;
+    let element = el;
+    while (element?.parentElement) {
+      const ariaHidden = element?.getAttribute('aria-hidden');
+      if (ariaHidden) return ariaHidden === 'true';
```

Comment:
This seems way too specific to navigation and the logic can very easily break with other elements. Not sure if this is the best solution for this issue. Should we look at a logic that checks if the element is in viewport or not? Maybe we can use `IntersectionObserver` API here. @christianvogt WDYT?

Reply:
While specific, this solution will be sufficient for now as we have limited highlight use cases. Since we call scrollIntoView, we cannot rely on a bounds check until the scrolling is complete either.

Please check `if (ariaHidden === 'true') return ...`.

Reply:
updated